Source/JavaScriptCore/runtime/FunctionPrototype.cpp

```diff
@@ -102,6 +103,105 @@ JSC_DEFINE_HOST_FUNCTION(functionProtoFuncToString, (JSGlobalObject* globalObjec
     return throwVMTypeError(globalObject, scope);
 }
 
+JSC_DEFINE_HOST_FUNCTION(functionProtoFuncBind, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSValue thisValue = callFrame->thisValue();
+    if (UNLIKELY(!thisValue.isCallable()))
+        return throwVMTypeError(globalObject, scope, "|this| is not a function inside Function.prototype.bind"_s);
+    JSObject* target = asObject(thisValue);
+
+    JSValue boundThis = callFrame->argument(0);
+    unsigned argumentCount = callFrame->argumentCount();
+    unsigned numBoundArgs = 0;
+    JSImmutableButterfly* butterfly = nullptr;
+    if (argumentCount > 1) {
+        numBoundArgs = argumentCount - 1;
```

Comment:
Is this correct?  `callFrame->argumentCount()` already excludes `this`.

Reply:
Correct. Since `bind`'s first argument is `|this|`